### PR TITLE
Document our Release process

### LINKED ISSUE #266: Summarize release process
So that the process is clear, summarize the steps need to release an updated SPF version.  Perhaps in a `RELEASING.md` file?

### PR DESCRIPTION
**Pull request by nicksay**
Closes #266.

### CODE REVIEW

**Comment by nicksay** [comment]
Updated to split changes to `CONTRIBUTING.md` into #284.

**Comment by awbraunstein** [comment]
How does pushing updated api documentation to the site work into this?

**Comment by nicksay** [comment]
Good point, I'll add a section about that.

As for formatting, which version do you think is most legible?
This:  https://raw.githubusercontent.com/nicksay/spfjs/doc-releasing/RELEASING.md
Alternate: https://raw.githubusercontent.com/nicksay/spfjs/doc-releasing-alt/RELEASING.md
Comparison: https://github.com/nicksay/spfjs/compare/nicksay:doc-releasing...doc-releasing-alt?expand=1

**Comment by awbraunstein** [comment]
I like the alternate. The whitespace makes it a lot easier on the eyes.

**Comment by nicksay** [comment]
PTAL

Updated the whitespace, added sections on documenting and announcing a release, and tweaked the other sections.

File: RELEASING.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+
+## Prepare the new version
+
+1.  Ensure all code has been tested. Verify changes from the
+    previous version manually and make sure all unit tests pass.
+
+2.  Update the `version` property in the `package.json` and
+    `bower.json` files to increment the number using
+    [semantic versioning][].
```
Comment (awbraunstein) [inline RELEASING.md]: This wording is a little odd. How about: 

Increment the `version` property in the `package.json` and `bower.json` files using semantic versioning.

Comment (nicksay) [inline RELEASING.md]: Sounds good, done.

**Comment by nicksay** [comment]
FYI I left the part of this as just "push the website".  I'll add a website-specific README under the web folder separately.

**Comment by awbraunstein** [comment]
lgtm :+1:

**Pull request merged**